.github/workflows/main.yml

```
@@ -95,6 +95,22 @@ jobs:
         env:
           CI_COMMIT_SHA: ${{ github.event.pull_request.head.sha || github.sha }}
 
+  test-ssr:
+    needs: setup
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Run Node.js ssr script
```

We should be setting up Node.js first, otherwise the system version on the GitHub Action runner will be used.
```suggestion
      - name: Set up Node
        uses: actions/setup-node@v4
        with:
          node-version: ${{ env.NODE_VERSION }}

      - name: Run Node.js ssr script
```